translated_mock_driver_testing/CMakeLists.txt

```diff
@@ -0,0 +1,139 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+cmake_minimum_required(VERSION 2.8.3)
+project(translated_mock_driver_testing)
+
+## Compile as C++11, supported in ROS Kinetic and newer
+add_compile_options(-std=c++11)
+set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
+set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
+
+# Find catkin macros and libraries
+## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
+## is used, also find other catkin packages
+find_package(catkin REQUIRED COMPONENTS
+  cav_msgs
+  cav_srvs
+  roscpp
+)
+
+###################################
+## catkin specific configuration ##
+###################################
+## The catkin_package macro generates cmake config files for your package
+## Declare things to be passed to dependent projects
+## INCLUDE_DIRS: uncomment this if your package contains header files
+## LIBRARIES: libraries you create in this project that dependent projects also need
+## CATKIN_DEPENDS: catkin_packages dependent projects also need
+## DEPENDS: system dependencies of this project that dependent projects also need
+catkin_package(
+#  LIBRARIES ui_integration
+   CATKIN_DEPENDS cav_msgs cav_srvs roscpp 
+)
+#  DEPENDS system_lib
+
+
+###########
+## Build ##
+###########
+
+## Specify additional locations of header files
+## Your package locations should be listed before other locations
+include_directories(
+  include
+  ${catkin_INCLUDE_DIRS}
+)
+
+
+## Declare a C++ executable
+## With catkin_make all packages are built within a single CMake context
+## The recommended prefix ensures that target names across packages don't collide
+file(GLOB_RECURSE headers */*.hpp */*.h)
+add_executable(${PROJECT_NAME}_node
+  src/main.cpp
+  src/MockLightBarDriver.cpp
+)
+
+add_library(${PROJECT_NAME}_lib
+                src/main.cpp
+                src/MockLightBarDriver.cpp
+                
+)
+
+
+
+## Add cmake target dependencies of the executable
+add_dependencies(${PROJECT_NAME}_node 
+                ${catkin_EXPORTED_TARGETS}
+)
+
+add_dependencies(${PROJECT_NAME}_lib
+                ${catkin_EXPORTED_TARGETS}
+) 
+
+
+## Specify libraries to link a library or executable target against
+target_link_libraries(${PROJECT_NAME}_node
+  ${catkin_LIBRARIES}
+)
+
+target_link_libraries(${PROJECT_NAME}_lib
+  ${catkin_LIBRARIES}
+)
+
+
+#############
+## Install ##
+#############
+
+## Mark cpp header files for installation
+install(DIRECTORY include/${PROJECT_NAME}
+DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+FILES_MATCHING PATTERN "*.hpp"
+PATTERN ".svn" EXCLUDE
+)
+
+## Mark executables and/or libraries for installation
+install(TARGETS ${PROJECT_NAME}_node
```

Comment:
So we should add ${PROJECT_NAME}_lib in here as we separated cpp files from lib and node